### PR TITLE
Uninstall config

### PR DESCRIPTION
**Pull request by SourceBolt**
Added Required parameters so you can't have an empty username and team name. Also added the feature to specify a desired timeout in milliseconds.

Screen Shot 2021-10-05 at 9 11 52 AM

Added the endpoint and timeout to the config file.

Screen Shot 2021-10-05 at 10 00 07 AM

Added the ability to uninstall from the installer binary
Screen Shot 2021-10-05 at 10 02 30 AM

If the config doesn't exist or is invalid the hook auto-uninstalls
Screen Shot 2021-10-05 at 9 56 50 AM

### CODE REVIEW

**Comment by AndreasChristianson** [comment]
I looked through this with the hope of understanding how uninstalls work for individual repos. It looks like the config is deleted then the hooks delete themselves. Am I understanding that correctly?

**Comment by AndreasChristianson** [comment]
I'll pull this down tonight and give it a try

**Comment by SourceBolt** [comment]
> I looked through this with the hope of understanding how uninstalls work for individual repos. It looks like the config is deleted then the hooks delete themselves. Am I understanding that correctly?

Yeah the description text hopefully outlines it but what the uninstall does currently on the install script is just unset the global flag as well as remove the config directory.

Each post-commit script that tries to get the config, deletes itself currently.

**Comment by AndreasChristianson** [comment]
For those upgrading, delete your current `~/.sai_git_hook_config` in order to avoid the following error
```
[Git-Hook-Installer]: What is your Team Name? Blog
[Git-Hook-Installer]: What is your Username? Andreas
[Git-Hook-Installer]: Desired Timeout in milliseconds (Default 3000 ms)?
[Git-Hook-Installer]: Not a directory (os error 20)
[Git-Hook-Installer]: Should this hook be installed globally? (Y|N) n
```

**Review by AndreasChristianson — DISMISSED** [review]
worked well for me

File: Hookstaller/src/post_commit.rs [hunk 1]
```diff
@@ -161,11 +159,31 @@ fn println_log<S: AsRef<str>>(output: S) {
...
     let config = match Config::read_config() {
         Ok(cfg) => cfg,
         Err(e) => {
             println_log(format!("Invalid Config: {}", e));
+            match uninstall_hook() {
+                Err(e) => {
+                    println_log(e.to_string());
+                }
+                Ok(_) => {
+                    println_log("Uninstalled Hook to reinstall run the install again");
```
Comment (AndreasChristianson) [inline Hookstaller/src/post_commit.rs:184]:
```suggestion
                    println_log("Successfully uninstalled Hook. To reinstall run install.sh again");
```

**Review by AndreasChristianson — APPROVED** [review]

**Pull request merged**